src/app/Scenes/Search/UserPrefsModel.tsx

```diff
@@ -43,19 +49,44 @@ export const getUserPrefsModel = (): UserPrefsModel => ({
     } else {
       console.warn("Currency not supported")
     }
-
-    // TODO: update currency in gravity
   }),
   setMetric: action((state, metric) => {
     if (metrics.includes(metric)) {
       state.metric = metric
     } else {
       console.warn("Metric/Dimension Unit not supported")
     }
+  }),
+  fetchRemoteUserPrefs: thunk(async (actions) => {
+    const me = await fetchMe()
 
-    // TODO: update unit in gravity
+    actions.setMetric(me?.lengthUnitPreference.toLowerCase() as Metric)
+    actions.setCurrency(me?.currencyPreference as Currency)
```

Comment:
We don't want to set the values if the query fails or returns null.

```suggestion
    const me = await fetchMe()

    if(!me) return

    actions.setMetric(me.lengthUnitPreference.toLowerCase() as Metric)
    actions.setCurrency(me.currencyPreference as Currency)
```